feat: replace PR 469 with the fix of failing memcheck test

maqi opened the pull request:
## Description

<!-- reviewpad:summarize:start -->
### Summary generated by Reviewpad on 06 Jul 23 10:22 UTC
This pull request includes the following changes:

1. Modifications to the `root_dir_path` function in `api.rs`, updating the comment to provide the correct file path for Windows operating systems.

2. Several changes in the `.github/workflows/nightly.yml` file related to environment variables and modified run commands.

3. Removal of the `init_logging` function and associated code in a file.

4. Changes in the `README.md` file, removing code and TODO items.

5. Changes in the `storage_payments.rs` file, including logging functionality and implementation of storage payment scenarios.

6. Modifications in a file related to logging configuration and other imports.

7. Changes in the `main.rs` file, including logging configuration updates and additional import statements.

These changes aim to improve logging functionality, update file paths, and make necessary modifications for proper execution and testing.
<!-- reviewpad:summarize:end -->

connec commented on .github/workflows/pr_tests.yml:
```diff
@@ -208,23 +208,29 @@ jobs:
-          taskkill /IM sn_node.exe /F
-          rd /s /q %USERPROFILE%\.safe\node
-        shell: cmd
+      # - name: Kill the current network (not needed for next test)
+      #   if: matrix.os != 'windows-latest'
+      #   run: |
+      #     killall sn_node
+      #     rm -rf ~/.safe/node
+      #   shell: bash
+
+      # - name: Kill the current network (not needed for next test)
+      #   if: matrix.os == 'windows-latest'
+      #   run: |
+      #     taskkill /IM sn_node.exe /F
+      #     rd /s /q %USERPROFILE%\.safe\node
+      #   shell: cmd
```
How come these are commented out?

joshuef replied:
Ah, cos the next test is commented out still also. I'll leave that out actually to tidyup the pr

connec reviewed (commented):
LGTM, one Q about some commented out stuff in the test workflow (also clippy ain't happy and some tests are failing, can't easily see which/why as the logs are too long).

connec commented:
In fact, MacOS is failing a new assertion: https://github.com/maidsafe/safe_network/pull/489/checks?check_run_id=3723267210

joshuef commented:
seems like there's a bug and we're still resending more on some AE instances than we'd want 🕵️

jacderida committed:
feat: provide `--log-output-dest` arg for `safenode`

BREAKING CHANGE: the `--log-dir` argument has been removed in favour of `--log-output-dest`.

This argument uses a 'special' value, which can be "stdout", "default", or a custom directory path.
The "default" is the default root directory, plus "logs" appended to that path.

The `sn_logging` crate now provides a `LogOutputDest` enum which makes for a nicer interface on the
logging initialisation, removing the need for an `Option<PathBuf>` to specify output to a file.

jacderida committed:
feat: provide `--log-output-dest` arg for `safe`

The behaviour for this is mostly the same as `safenode`, except if the argument is not
specified, no logging will take place (this does not affect println output).

jacderida committed:
feat: introduce `--log-format` arguments

BREAKING CHANGE: for both `safe` and `safenode` the `--json-log-output` argument is removed in
favour of this new `--log-format` argument.

The argument functions in a similar fashion to `--log-output-dest`, in that it uses a 'special'
value, which for the time being can either be "default" or "json". It has the potential to be
expanded.

jacderida committed:
feat: remove option from `--log-output-dest` arg

Benno submitted a patch that removes the need for `log_output_dest` to be expressed as an Option,
which makes the argument easier to work with.

jacderida committed:
chore: incorporate various feedback items

* Keep the logging guard alive for the scope of the whole program in `safe`
* Remove duplicate code in favour of function call
* Use the term "root-dir" rather than "default" to avoid confusion with the fact that "stdout" is
  actually the default value

jacderida committed:
ci: update node logging paths

In light of the changes to the logging setup, the testnet tool is modified such that it will use the
default root and logging directories.

The merge workflow is also updated to look for logs in the new directories. For the correct
usage of paths, in some cases this meant using different mechanisms on Windows vs. Unix.

The memcheck workflow was also updated to the same effect, but it only runs on Linux, so
cross-platform considerations weren't required.

jacderida committed:
chore: use data-dir rather than root-dir

Completely separate the `--log-output-dest` argument from the `--root-dir` argument by removing the
use of the term "root-dir" from the logging output destinations, which could have been confusing.

jacderida committed:
ci: update benchmark workflows for new directories

This was missed in the previous commit, but is now updated in the same fashion as the merge and
memcheck workflows.

jacderida committed:
ci: use variables for heaptrack directory

This is both for readability and maintainability and matches the way the other directories are
referenced.

jacderida committed:
tests: clear out chunks and registers

The churn tests were making an assumption that the log and root directories were the same, but this
no longer holds.

The response doesn't yet contain the root directory, so we can infer that it's one above the logging
directory, which is now underneath the peer ID root directory.

joshuef committed:
chore: tidy remove_file call

joshuef committed:
feat: add restart func for node process

Restart now closes the current process down, ending the runtime and
spawns a new process, using the original bin and args passed
in to this one.

joshuef committed:
chore: temp change to memcheck folder counting

joshuef committed:
ci: standardise heaptrack mem consumption to always be in mbs

joshuef committed:
ci: log the client upload zip file size

This should help see if client mem spikes are build folder size releated

joshuef committed:
ci: add node data path to churn matrix

joshuef committed:
ci: tweak client mem limit

joshuef committed:
ci: fix benchmark heaptrack conversions

jacderida committed:
ci: use `heaptrack` to launch `safe` in correct way

It looks like when you pass arguments to a process launched with heaptrack, optional arguments need
to come before positional arguments.

Some style changes were made here to some steps with Bash scripts. A few steps declared environment
variables when they can just be variables of the script, since no other process needs access to
them, and other variable references were changed to lowercase to distinguish them from constant
values.

maqi committed:
fix: use SAFE_PEERS as fall back initial peers for non-local-discovery
